src/domain/interfaces/counterfactual-safes.datasource.interface.ts

```diff
@@ -13,6 +13,7 @@ export interface ICounterfactualSafesDatasource {
   }): Promise<CounterfactualSafe>;
 
   getCounterfactualSafe(args: {
+    account: Account;
```

Comment:
I've added the `account`  here, as it's always available on the domain level, and it narrows down the query surface, improving performance.